tutorials/nservicebus-101/lesson-4/tutorial.md

```diff
@@ -0,0 +1,212 @@
+
+Let's take a look at all of these differences side-by-side:
+
+|   | Commands | Events |
+|---|:--------:|:------:|
+| Marker Interface | `ICommand` | `IEvent` |
+| Logical Senders | Many | 1 |
+| Logical Receivers | 1 | Many (or none) |
+| Purpose | "Please do something" | "Something has happened" |
+| Naming (Tense) | Imperative | Past |
+| Examples | `PlaceOrder`<br/> `UpdateCustomerStatus`<br/>`ChargeCreditCard` | `OrderPlaced`<br/>`CustomerStatusUpdated`<br/>`CreditCardCharged` |
+| Coupling Style | Tight | Loose |
+
+From this comparison, it's clear that commands and events will sometimes come in pairs. A command will arrive, perhaps from a website UI, telling the system to `DoSomething`. The system does that work, and as a result, publishes a `SomethingHappened` event, which other components in the system can react to.
+
+The loose coupling provided by publishing events gives us quite a bit of flexibility to design our software systems in a much more maintainable way.
+
+
+## Better code through decoupling
+
+Have you ever seen a codebase for a naive implementation of an e-commerce store? Commonly you'll find one gargantuan SubmitOrder method responsible for retrieving the shopping cart, creating an Order and OrderLines in the database, authorizing the credit card, capturing the credit card authorization, emailing a receipt, notifying a fulfillment agency, and then updating any sort of wish list, gift registry, or "frequently bought together" information the site might keep track of. If that method doesn't number in the hundreds of lines of code, then it likely calls into subroutine methods to handle each of these concerns. The sum total lines of code in such an example likely numbers in at least hundreds of lines of code, if not thousands.
+
+What happens when one of the steps in that long chain fails? You're left with a partially completed process that requires manual intervention to fix, either by mucking around manually in the database, manually reconciling with a credit card processor, or manually sending confirmation emails.
+
+By using events, we can better follow the [single responsibility principle](https://en.wikipedia.org/wiki/Single_responsibility_principle) and divide up these concerns into separate message handlers. Simply publish `OrderPlaced`, and all the other components that subscribe to it will take care of their own concerns.
+
+This means that when the code for the credit card processing changes, we don't even need to touch (let alone test and redeploy) any of the code in the system except for that which is directly related to processing credit cards.
+
+
+## Defining events
+
+Creating an event message is just as easy as creating an event. We just create a class and mark it with the `IEvent` (rather than `ICommand`) marker interface.
```

Comment:
Creating an event message similar to creating an event. We just create a class and mark it with the `IEvent` (rather than `ICommand`) marker interface.